Example-Crypto-REXX/RXCRYPT1.rexx

```diff
@@ -0,0 +1,2453 @@
+
+      CPCname = word(parmString, i)
+      localCPC = FALSE
+      i = i + 1
+    end
+  else if TRANSLATE(localArg) == '-S' then
+    do /* -S <Status> */
+      i = i + 1
+      if i > argCount then
+       return usage('-S option specified, but is missing Status')
+
+      ReqstdLPAR_Status = word(parmString, i)
+      i = i + 1
+    end
+  else if TRANSLATE(localArg) == '-D' then
+    do /* -D <data set name> */
+      i = i + 1
+      if i > argCount then
+       return usage('-D option specified, but is missing the data set name')
+
+      TEMP_DATASET = word(parmString, i)
+      dataSetProvided = TRUE
+      i = i + 1
+    end
+  else
+    do
+      errMsg = 'Found unsupported argument:'||localArg
+      return usage(errMsg)
+    end
+ end /* while loop */
+
+if dataSetProvided = FALSE then
+  return usage ('Missing required data set name for output')
+
+return 0  /* end function */
+
+/***********************************************/
+/* Function: VerifyStatus                      */
+/*                                             */
+/* Verify the requested LPAR status            */
+/* If invalid, set it to 'operating'           */
+/*                                             */
+/* Returns: 0 if successful                    */
+/*         -1 if not successful                */
+/***********************************************/
+VerifyStatus:
+
+statusString = arg(1)
+LPAR_STATUS_NOT_ACTIVATED = TRANSLATE('not-activated')
+LPAR_STATUS_NOT_OPERATING = TRANSLATE('not-operating')
+LPAR_STATUS_OPERATING = TRANSLATE('operating')
+
+Select
+ When TRANSLATE(statusString) = LPAR_STATUS_NOT_ACTIVATED
+  then say 'LPAR status '||statusString||' will be used'
+ When TRANSLATE(statusString) = LPAR_STATUS_NOT_OPERATING
+  then say 'LPAR status '||statusString||' will be used'
+ When TRANSLATE(statusString) = LPAR_STATUS_OPERATING
+  then say 'LPAR status '||statusString||' will be used'
+ OTHERWISE
+  do
+   call EntryArrow
+   say 'Invalid LPAR status entered, default of operating will be used'
+   ReqstdLPAR_Status = 'operating'
+   call ExitArrow
+  end
+end
+
+return 0  /* end function */
+
+/***********************************************/
+/* Function: usage                             */
+/*                                             */
+/* Provide usage guidance to the invoker.      */
+/*                                             */
+/* Returns: -1 to indicate fatal script error. */
+/***********************************************/
+usage:
+ whyString = arg(1)
+ say
+ say 'USAGE: RXCRYPT1 -D outputDataSet [-C CPCname] [-I] [-V]'
+ say '    REQUIRED'
+ say '         -D outputDataSet, name of an existing partitioned'
+ say '               data set, if the LPAR query is successful, a member '
+ say '               containing the query information in CSV format will '
+ say '               be stored into the data set, the member will either '
+ say '               be named LOCAL or the CPC name specified via -C option'
```

Comment:
needs a little update to specify crypto stuff not lpar stuff, would you also double check the other -D descriptions

Reply:
fixed